ott/core/sinkhorn_lr.py

```diff
@@ -496,22 +607,39 @@ def one_iteration(
     else:
       state = self.kernel_step(ot_prob, state, iteration)
 
+    # compute the criterion
+    criterion = state.compute_crit()
```

Comment:
See my comment re `LRSinkhornState` refactoring. Here I imagine it would be like:
```python
...
new_state = self.lse_step(ot_prob, state, iteration)
criterion = new_state.compute_criterion(old_state)
criterions = state.criterions.at[iteration // self.inner_iterations].set(criterion)
...

```